lib/pages/saved_places.dart

```diff
@@ -0,0 +1,374 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:latlong/latlong.dart';
+
+import '../blocs/saved_locations_bloc.dart';
+
+import '../trufi_configuration.dart';
+import '../trufi_localizations.dart';
+import '../trufi_models.dart';
+import '../widgets/trufi_drawer.dart';
+
+import 'choose_location.dart';
+
+class SavedPlacesPage extends StatefulWidget {
+  static const String route = '/places';
+
+  @override
+  State<StatefulWidget> createState() => SavedPlacesPageState();
+}
+
+class SavedPlacesPageState extends State<SavedPlacesPage> {
+  LatLng _center;
+  TextEditingController textController = TextEditingController();
+
+  @override
+  void dispose() {
+    textController.dispose();
+    super.dispose();
+  }
+
+  final Map<String, IconData> icons = <String, IconData>{
+    'saved_place:home': Icons.home,
+    'saved_place:work': Icons.work,
+    'saved_place:fastfood': Icons.fastfood,
+    'saved_place:local_cafe': Icons.local_cafe,
+    'saved_place:map': Icons.map,
+  };
+
+  @override
+  void initState() {
+    super.initState();
+    _loadState();
+  }
+
+  void _loadState() {
+    final SavedLocationsBloc savedLocationsBloc =
+        SavedLocationsBloc.of(context);
+    final TrufiConfigurationMap map = TrufiConfiguration().map;
+    final List<TrufiLocation> data =
+        savedLocationsBloc.locations.reversed.toList();
+
+    final List<String> descriptions = data.map((TrufiLocation location) {
+      return location.description;
+    }).toList();
+
+    if (!descriptions.contains('Home')) {
```

Comment:
does this prevent to have more than one home / work ? Are you sure you want to use the description for this check? maybe type is better?

Reply:
No, it does not. I think is better have a boolean in Shared_preferences to fix it.

Reply:
Thank you.

Reply:
I agree this bit won't work too well.

1) When I remove Home or Work, it will be back the next time I open the view.
2) The placement of this "initialization" code in the page view is not ideal. It will show no entries the first time I open Your Places, and silently add them. And then only the second time I open the view will Work and Home be visible. The initialization should be done in the bloc.

Reply:
Oh perfect, I understand.